staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -375,6 +398,43 @@ func TestWatchCacheBypass(t *testing.T) {
 	}
 }
 
+func TestTooManyRequestsNotReturned(t *testing.T) {
+	// Ensure that with ResilientWatchCacheInitialization feature disabled, we don't return 429
+	// errors when watchcache is not initialized.
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ResilientWatchCacheInitialization, false)
+
+	dummyErr := fmt.Errorf("dummy")
+	backingStorage := &dummyStorage{err: dummyErr}
+	cacher, _, err := newTestCacherWithoutSyncing(backingStorage)
+	if err != nil {
+		t.Fatalf("Couldn't create cacher: %v", err)
+	}
+	defer cacher.Stop()
+
+	opts := storage.ListOptions{
+		ResourceVersion: "0",
+		Predicate:       storage.Everything,
+	}
+
+	// Cancel the request so that it doesn't hang forever.
+	listCtx, listCancel := context.WithTimeout(context.Background(), 250*time.Millisecond)
+	defer listCancel()
+
+	result := &example.PodList{}
+	err = cacher.GetList(listCtx, "/pods/ns", opts, result)
+	if err != nil && apierrors.IsTooManyRequests(err) {
```

Comment:
here we expect the error to be exactly `dummyErr`

Reply:
No - dummyErr is what watchcache get, it's not piping this error, it just doesn't allow watchcache to initialized.

Reply:
mhm, I think that the request [will go](https://github.com/kubernetes/kubernetes/blob/a8ef6e9f0104a44023162bb8229fb677ec80beb1/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L811) to etcd and the error will be returned.